homeassistant/components/pi_hole/__init__.py

```diff
@@ -31,6 +40,7 @@
             {
                 vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
                 vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+                vol.Optional(CONF_API_KEY, default=None): cv.optional_string,
```

Comment:
Please remove the default and use `cv.string` as validator.

We use `dict.get` to get the config option so there's no need for a default in this case.

Reply:
I think there is still a need for a default, albeit elsewhere. Otherwise, a `KeyError` would be thrown when trying to retrieve it.

Reply:
`dict.get` will return `None` if the key is missing in the dict.

Reply:
Right, I thought the preferred syntax was `dict[key]`

Reply:
Yes, when we know that the key is in the dict. In this case it's optional so we don't know that the key is in the dict. Then we use `dict.get`.